rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/action/TbCreateAlarmNode.java

```diff
@@ -160,4 +162,22 @@ private Alarm buildAlarm(TbMsg msg, JsonNode details, TenantId tenantId) {
                 .build();
     }
 
+    private AlarmSeverity processAlarmSeverity(TbMsg msg) {
+        AlarmSeverity alarmSeverity;
+        if(this.config.isDynamicSeverity()) {
+            String processPattern = TbNodeUtils.processPattern(this.config.getSeverity(), msg);
+            if(!EnumUtils.isValidEnum(AlarmSeverity.class, processPattern)) {
+                alarmSeverity = null;
+            } else {
+                alarmSeverity = EnumUtils.getEnum(AlarmSeverity.class, processPattern);
+            }
+        } else {
+            alarmSeverity = EnumUtils.getEnum(AlarmSeverity.class, this.config.getSeverity());
```

Comment:
No need to execute  EnumUtils.getEnum(AlarmSeverity.class, this.config.getSeverity()) if you are not using the dynamicSeverity. Execute this once in the init method and throw an exception if the value is invalid!